Source/player.cpp

```diff
@@ -600,12 +600,12 @@ bool PlrHitMonst(Player &player, Monster &monster, bool adjacentDamage = false)
 	}
 
 	if (gbIsHellfire && HasAllOf(player._pIFlags, ItemSpecialEffect::FireDamage | ItemSpecialEffect::LightningDamage)) {
-		int midam = player._pIFMinDam + GenerateRnd(player._pIFMaxDam - player._pIFMinDam);
+		int midam = RandomIntBetween(player._pIFMinDam, player._pIFMaxDam);

```

Comment:
This and the change on line 923 fix off-by-one bugs as well, not sure if you want to call that out in comments. (if that's unintended then pass `true` as the third arg to randomintbetween to use half-open ranges)

Reply:
Will add bugfix comments